Sources/Java2SwiftLib/JavaTranslator+Validation.swift

```diff
@@ -0,0 +1,72 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift.org project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift.org project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+package extension JavaTranslator {
+  struct SwiftTypeName: Hashable {
+    let swiftType: String
+    let swiftModule: String?
+
+    package init(swiftType: String, swiftModule: String?) {
+      self.swiftType = swiftType
+      self.swiftModule = swiftModule
+    }
+  }
+
+  struct SwiftToJavaMapping: Equatable {
+    let swiftType: SwiftTypeName
+    let javaTypes: [String]
+
+    package init(swiftType: SwiftTypeName, javaTypes: [String]) {
+      self.swiftType = swiftType
+      self.javaTypes = javaTypes
+    }
+  }
+
+  enum ValidationError: Error, CustomStringConvertible {
+    case multipleClassesMappedToSameName(swiftToJavaMapping: [SwiftToJavaMapping])
+
+    package var description: String {
+      switch self {
+      case .multipleClassesMappedToSameName(let swiftToJavaMapping):
+              """
+              The following Java classes were mapped to the same Swift type:
```

Comment:
```suggestion
              The following Java classes were mapped to the same Swift type name:
```

include "name" maybe? It made it sound like we mapped to an existing type but the problem is we didn't form a type for it at all since the conflict